stretch_core/stretch_core/joint_trajectory_server.py

```diff
@@ -204,3 +215,57 @@ def success_callback(self, success_str):
         self.result.error_code = self.result.SUCCESSFUL
         self.result.error_string = success_str
         self.goal_handle.succeed()
+
+    def execute_trajectory(self, goal_handle):
+        try:
+            with self.node.robot_stop_lock:
+                # Escape stopped mode to execute trajectory
+                self.node.stop_the_robot = False
+                self.node.robot_mode_rwlock.acquire_read()
+
+            # Process the goal
+            goal = goal_handle.request
+
+            if self.ignore_trajectory_velocities or self.ignore_trajectory_accelerations:
+                for pt in goal.trajectory.points:
+                    if self.ignore_trajectory_velocities:
+                        pt.velocities = []
+                    if self.ignore_trajectory_accelerations:
+                        pt.accelerations = []
+
+            # Print the goal
+            if goal.trajectory.points:
+                dt = to_sec(goal.trajectory.points[-1].time_from_start)
+                n_points = len(goal.trajectory.points)
+            else:
+                dt = 0.0
+                n_points = 0
```

Comment:
Can the robot get a goal trajectory without any points? Is this normal? If not rather than printing 0 points an appropriate error message would be better imo.

Reply:
This is actually placeholder for the case where you only have MultiDOF points.

Reply:
I see that's OK then.